core/integration/module_test.go

```diff
@@ -2286,6 +2286,192 @@ type Test struct {}
+// having to implements the full interface but only the ones you want.
+// cc: https://github.com/dagger/dagger/issues/7707
+func (ModuleSuite) TestUnbundleSDK(ctx context.Context, t *testctx.T) {
+	t.Run("only codegen", func(ctx context.Context, t *testctx.T) {
+		c := connect(ctx, t)
+
+		ctr := c.Container().From(golangImage).
+			WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+			WithWorkdir("/work/sdk").
+			With(daggerExec("init", "--name=only-codegen", "--sdk=go", "--source=.")).
+			WithNewFile("/work/sdk/main.go", `package main
+
+import (
+  "context"
+  "dagger/only-codegen/internal/dagger"
+)
+
+type OnlyCodegen struct{}
+
+func (o *OnlyCodegen) Codegen(
+  ctx context.Context,
+  modSource *dagger.ModuleSource,
+  introspectionJSON *dagger.File,
+) (*dagger.GeneratedCode, error) {
+  return dag.GeneratedCode(dag.Directory().WithNewFile("hello.txt", "Hello, world!")), nil
+}
+`).
+			WithWorkdir("/work").
+			With(daggerExec("init", "--name=test", "--sdk=./sdk", "--source=."))
+
+		t.Run("can run dagger develop", func(ctx context.Context, t *testctx.T) {
+			generatedFile, err := ctr.With(daggerExec("develop")).File("/work/hello.txt").Contents(ctx)
+
+			require.NoError(t, err)
+			require.Equal(t, "Hello, world!", generatedFile)
+		})
+
+		t.Run("explicit error on dagger call", func(ctx context.Context, t *testctx.T) {
+			_, err := ctr.With(daggerExec("call", "foo")).Sync(ctx)
+
+			requireErrOut(t, err, "the SDK does not implement the `Runtime` interface")
+		})
+
+		t.Run("explicit error on dagger functions", func(ctx context.Context, t *testctx.T) {
+			_, err := ctr.With(daggerFunctions()).Sync(ctx)
+
+			requireErrOut(t, err, "the SDK does not implement the `Runtime` interface")
+		})
+	})
+
+	t.Run("only runtime", func(ctx context.Context, t *testctx.T) {
+		c := connect(ctx, t)
+
+		ctr := c.Container().From(golangImage).
+			WithMountedFile(testCLIBinPath, daggerCliFile(t, c)).
+			WithWorkdir("/work/sdk").
+			With(daggerExec("init", "--name=only-runtime", "--sdk=go", "--source=.")).
+			WithNewFile("/work/sdk/main.go", fmt.Sprintf(`package main
+
+import (
+  "context"
+  "dagger/only-runtime/internal/dagger"
+)
+
+type OnlyRuntime struct {
+  Src *dagger.Directory
+}
+
+func New(
+  //+defaultPath="./src"
+  sdkSrc *dagger.Directory,
+) *OnlyRuntime {
+  return &OnlyRuntime{
+    Src: sdkSrc,
+  }
+}
+
+func (m *OnlyRuntime) ModuleRuntime(
+  ctx context.Context,
+  modSource *dagger.ModuleSource,
+  introspectionJSON *dagger.File,
+) (*dagger.Container, error) {
+  return dag.Container().
+    From("%s").
+    WithDirectory("/src", m.Src).
+    WithWorkdir("/src").
+    WithExec([]string{"go", "build", "-o", "/bin/sdk", "."}).
+    WithEntrypoint([]string{"/bin/sdk"}), nil
+}
+`, golangImage)).
+			WithWorkdir("/work/sdk/src").
+			WithExec([]string{"go", "mod", "init", "dagger.test/sdk"}).
+			WithExec([]string{"go", "get", "dagger.io/dagger@v0.18.9"}).
+			WithNewFile("/work/sdk/src/main.go", `package main
```

Comment:
I think it makes sense to move these little custom SDKs out into a dedicated go file, and `embed` it.

Reply:
I'm not sure I can, it will fails on the lint because the module imports dagger types that needs to be generated? 
Is there a way to embed and copy the whole directory inside the engine?